userlist interface

kawyua opened the pull request:
Edit: new gif of user list demonstration
![userlist demonstration3](https://user-images.githubusercontent.com/12496046/69468457-383d9880-0d5a-11ea-80ea-e828328e0157.gif)

This is currently what I have. Any thoughts on what I should do?

Addressed all of shibins comments.
Edit2: added unsubscribe and removed getUserByEmail stub. We'll have to figure out specifically how adding user to chatroom will work.
edit3: moved unsubscribe function inside getUserList

shiBinH commented on src/app/chatbox/chatbox.component.ts:
```diff
@@ -81,6 +99,7 @@ export class ChatboxComponent implements OnInit {
           this.openConversation(0);
         }
       });
+    this.updateUserList();
```
There's a race condition here. `updateUserList()` should be called after `openConversation(0)` completes because `openConversation(0)` selects the first (0th) chatroom as the currently opened chatroom and then loads its chat history. `updateUserList()` should only display the users in the currently opened chatroom, and that only happens if `openConversation(0)` completes before `updateUserList()`.  `openConversation()` may have to be modified to return a `Promise` so then it can be chained into loading the user list.

kawyua commented on src/app/chatbox/chatbox.component.ts:
```diff
@@ -132,42 +151,71 @@ export class ChatboxComponent implements OnInit {
+  }
+
+  updateUserList() {
+    this.userListSubscription = this.userInfoService
```
oh wow that is where the unsubscribe for chatroom was. So placing this statement:
 ```
   if (this.userListSubscription) {
      this.userListSubscription.unsubscribe();
    }
```
will fix the memory leak then. made a new commit with all these changes

Hongzred commented:
the ui stills needs to be improved, but the functionality looks good

kawyua commented:
> 
> 
> Let's ship this and look into adding tests in a future commit. You can rebase onto master and then merge.

yeah sorry I haven't made much progress in unit tests yet. ill get it done soon. will merge addressing your comments
